python/tests/test_parsimony.py

```diff
@@ -1250,6 +1258,69 @@ def test_reversed_mutation_over_7(self):
         assert transitions[0] == tskit.Mutation(node=7, derived_state="0")
 
 
+class TestParsimonyExamplesAncestralState(TestParsimonyBase):
+    """
+    Test fixing the ancestral state. Note that a mutation can occur above node 10
+    to switch the ancestral state
+    """
+
+    #     10
+    #    ┏━┻━┓
+    #    ┃   9
+    #    ┃ ┏━┻━┓
+    #    ┃ ┃   8
+    #    ┃ ┃ ┏━┻━┓
+    #    ┃ ┃ ┃   7
+    #    ┃ ┃ ┃ ┏━┻┓
+    #    ┃ ┃ ┃ ┃  6
+    #    ┃ ┃ ┃ ┃ ┏┻┓
+    #    0 1 2 3 4 5
+    tree = tskit.Tree.generate_comb(6)
+
+    def test_mutation_over_0(self):
+        genotypes = [1, 0, 0, 0, 0, 0]
+        ancestral_state, transitions = self.do_map_mutations(
+            self.tree, genotypes, ancestral_state=0, compare_lib=False
+        )
+        assert ancestral_state == "0"
+        assert len(transitions) == 1
+        assert transitions[0] == tskit.Mutation(node=0, parent=-1, derived_state="1")
+        ancestral_state, transitions = self.do_map_mutations(
+            self.tree, genotypes, ancestral_state=1, compare_lib=False
+        )
+        assert ancestral_state == "1"
+        assert len(transitions) == 1
+        assert transitions[0] == tskit.Mutation(node=9, parent=-1, derived_state="0")
+
+    def test_mutation_over_3(self):
+        genotypes = [0, 0, 0, 1, 0, 0]
+        ancestral_state, transitions = self.do_map_mutations(
+            self.tree, genotypes, ancestral_state=tskit.MISSING_DATA, compare_lib=False
+        )
+        assert ancestral_state == "0"
+        assert len(transitions) == 1
+        assert transitions[0] == tskit.Mutation(node=3, parent=-1, derived_state="1")
+        genotypes = [0, 0, 0, 1, 0, 0]
```

Comment:
Helpful to put in a blank like to separate these different tests visually

Reply:
Good point